data_model/src/isi.rs

```diff
@@ -1508,6 +1378,18 @@ pub mod error {
             AssetDefinitionId(
                 #[cfg_attr(feature = "std", source)] Box<Mismatch<AssetDefinitionId>>,
             ),
+            /// Numeric asset value type was expected, received: {0}
+            NumericAssetValueTypeExpected(
+                #[skip_from]
+                #[skip_try_from]
+                AssetValueType,
+            ),
+            /// Store asset value type was expected, received: {0}
+            StoreAssetValueTypeExpected(
+                #[skip_from]
+                #[skip_try_from]
+                AssetValueType,
+            ),
```

Comment:
these seem redundant since we have `TypeError::AssetValueType`?

Reply:
No quite, because this error tells us that user send wrong instruction for the given `AssetDefinition` value type. 

While `TypeError::AssetValueType` is used when value type matches but specification is violated (like user send fractional decimal when only integers are accepted).

Reply:
They differ in amount of information available when throwing error.

`NumericAssetValueTypeExpected/StoreAssetValueTypeExpected` comparing `isi` and `asset_definition` value type.

`TypeError::AssetValueType` comparing `asset_definition` value type and `asset` value itself